acceptance/tests/environment/negative/agent_run_should_fail_if_env_unreadable.rb

```diff
@@ -0,0 +1,57 @@
+test_name "C97899 - Agent run should fail if environment is unreadable" do
+
+  testdir = ''
+  env_path = ''
+  test_env = ''
+
+  step 'setup environments' do
+    testdir = create_tmpdir_for_user master, 'c97899_unreadable_envdir'
+    env_path = "#{testdir}/environments"
+    test_env = "#{env_path}/testing"
+
+    apply_manifest_on(master, <<-MANIFEST, :catch_failures => true)
+      File {
+        ensure => directory,
+        mode => "0770",
+        owner => #{master.puppet['user']},
+        group => #{master.puppet['group']},
+      }
+      file {
+        '#{env_path}':;
+        '#{env_path}/production':;
+        '#{test_env}':;
+        '#{test_env}/manifests':;
+        '#{test_env}/modules':;
+      }
+      file { '#{test_env}/manifests/site.pp':
+        ensure => file,
+        mode => "0640",
+        content => 'node default { notify { "Hello agent": } }',
+      }
+    MANIFEST
+  end
+
+  step 'change permissions of envdir to 644' do
+    on(master, "chmod 644 #{test_env}")
```

Comment:
There's two possible cases here: the environment folder has overly restrictive permissions; and the environment folder manifest has correct permissions but site.pp has overly restrictive permissions.
This code seems to restrict both permissions but will only really be testing the folder permissions. Perhaps this test file (or a separate test file) should test the case of the folder having correct permissions and site.pp having restrictive permissions?

(I imagine we could also have cases where both the folder and site.pp have correct permissions but then other manifest files/folders are restricted - I think that should be a separate ticket though)